lib/logflare/scheduler.ex

```diff
@@ -2,5 +2,14 @@ defmodule Logflare.Scheduler do
   @moduledoc """
   Quantum scheduler for periodic tasks.
   """
-  use Quantum, otp_app: :logflare
+  use Quantum, otp_app: :logflare, restart: :transient
+
+  @doc """
+  Returns the scheduler :via name used for syn registry.
+  """
+  def scheduler_name do
+    ts = DateTime.utc_now() |> DateTime.to_unix(:nanosecond)
```

Comment:
DateTime.utc_now uses `System.os_time(:nanosecond)` to build the DateTime, so you can avoid the building and conversion by calling it directly:

```suggestion
    ts = System.os_time(:nanosecond)
```